Hunter/Assets/Scripts/View/View.cs

```diff
@@ -32,62 +33,73 @@ public class View : MonoBehaviour
     [TextArea]
     private string _deersParentName;
 
+    private void Awake()
+    {
+        _entityFactory = new EntityFactory();
+
+    }
+
     private void Start()
     {
         _game = new(_rabbitsNumber, _deersNumber, 0);
 
+        CreateEntityObjects();
         CreateEntities();
-        CreateEntitiesDeers();
     }
 
-    // TO-DO: use some generic type to create all entities using this
-    // Method (maybe use some enums)
-    private void CreateEntities()
+    private void CreateEntityObjects()
     {
-        GameObject rabbitsParent = new GameObject
-        {
-            name = _rabbitsParentName
-        };
-        rabbitsParent.transform.parent = transform;
-
-        foreach (Entity entity in _game.Rabbits)
-        {
-            GameObject entityObject = Instantiate(_rabbitsPrefab,
-                rabbitsParent.transform);
-
-            float xPosition = entity.Position.X;
-            float yPosition = entity.Position.Y;
-            entityObject.transform.localPosition =
-                new Vector3(xPosition, yPosition);
-
-            _entities.Add(entity, entityObject);
-        }
+        var _rabbitObject = new EntityObject(
+            AnimalType.Rabbit,
+            _rabbitsPrefab,
+            _rabbitsNumber
+        );
+        _entityFactory.AddEntityObject(_rabbitObject);
+
+        var _deerObject = new EntityObject(
+            AnimalType.Deer,
+            _deersPrefab,
+            _deersNumber * 10
+        );
+        _entityFactory.AddEntityObject(_deerObject);
+
+        _entityFactory.CreateEntityObjects();
     }
 
-    private void CreateEntitiesDeers()
+    private void CreateEntities()
     {
-        GameObject deersParent = new GameObject
+        foreach (AnimalType animalType in (AnimalType[])Enum.GetValues(typeof(AnimalType)))
         {
-            name = _deersParentName
-        };
-        deersParent.transform.parent = transform;
-
-        foreach (Herd herd in _game.HerdsOfDeer)
-        {
-            foreach (Entity entity in herd.GetAnimals())
+            //List<Animal> listOfAnimals = _game.Animals[animalType];
+            List<Animal> listOfAnimals = GetListOfAnimals(animalType);
+            foreach (Animal anim in listOfAnimals)
             {
-                GameObject entityObject = Instantiate(_deersPrefab,
-                deersParent.transform);
-
-                float xPosition = entity.Position.X;
-                float yPosition = entity.Position.Y;
-                entityObject.transform.localPosition =
-                    new Vector3(xPosition, yPosition);
+                GameObject entityObject = _entityFactory.GetEntity(animalType, anim);
 
-                _entities.Add(entity, entityObject);
+                _entities.Add(anim, entityObject);
             }
+        }
+    }
 
+    private List<Animal> GetListOfAnimals(AnimalType animalType)
+    {
+        List<Animal> listOfAnimals = new();
```

Comment:
as for Me listOfAnimals" is not a good name:
1. using "of", "a" and "the" in variables - excessive. If You replace first and last words sense won't change (listOfAnimals and animalsList)
2. this point comes out of the first one: don't use type of variable in name (like "list", "array", "int" and etc.)

Reply:
*I have this issue too (especially, when I write code from scratch)

Reply:
renamed, thanks